activerecord/lib/active_record/migration.rb

```diff
@@ -1013,10 +1033,15 @@ def record_version_state_after_migrating(version)
         ActiveRecord::SchemaMigration.where(:version => version.to_s).delete_all
       else
         migrated << version
-        ActiveRecord::SchemaMigration.create!(:version => version.to_s)
+        ActiveRecord::SchemaMigration.create!(version: version.to_s, environment: ActiveRecord::ConnectionHandling::DEFAULT_ENV.call)
       end
     end
 
+    def self.production?
+      raise NoEnvironmentInSchemaError unless SchemaMigration.environment_is_stored?
+      ActiveRecord::SchemaMigration.order(:created_at).last.environment == "production"
```

Comment:
Should we just compare to the current environment, rather than special casing production?